Lib/test/test_functools.py

```diff
@@ -1654,6 +1654,12 @@ def orig(x, y):
             with self.subTest(func=f):
                 f_copy = copy.deepcopy(f)
                 self.assertIs(f_copy, f)
+    def test_lru_cache_parameters(self):
+        def f():
+            return 1
+        f = self.module.lru_cache(2)(f)
```

Comment:
Let's use the decorator notation to create the example.  Also, make the example use `maxsize=1000` and `typed=True`.